File: cpp/src/Ice/ResponseHandler.cpp
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

#include <Ice/ResponseHandler.h>

using namespace std;
using namespace IceInternal;

IceUtil::Shared* IceInternal::upCast(ResponseHandler* obj) { return obj; }

ResponseHandler::~ResponseHandler()
{
}

File: csharp/src/Ice/ResponseHandler.cs
// **********************************************************************
//
// Copyright (c) 2003-present ZeroC, Inc. All rights reserved.
//
// **********************************************************************

using System.Collections.Generic;
using Ice.Instrumentation;

namespace IceInternal
{
    public interface ResponseHandler
    {
        void sendResponse(int requestId, Ice.OutputStream os, byte status, bool amd);
        void sendNoResponse();
        bool systemException(int requestId, Ice.SystemException ex, bool amd);
        void invokeException(int requestId, Ice.LocalException ex, int invokeNum, bool amd);
    }
}
